baseapp/baseapp.go

```diff
@@ -609,6 +609,141 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
+// runCheckTxConcurrently processes a transaction with either the checkTx or recheckTx modes, encoded transaction
+// bytes, and the decoded transaction itself. All state transitions occur through
+// a cached Context depending on the mode provided.
+//
+// Note, gas execution info is always returned. A reference to a Result is
+// returned if the tx does not run out of gas and if all the messages are valid
+// and execute successfully. An error is returned otherwise.
+func (app *BaseApp) runCheckTxConcurrently(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, result *sdk.Result, anteEvents []abci.Event, priority int64, err error) {
```

Comment:
In general, I think there are duplicate code between this new method and the `runTx` method.

Is there room to refactor out common components? Or is the goal here to keep the `runTx` unmodified to keep the rebasing easier in the future? Or something else?

How do we plan to keep this method in sync w future updates to `runTx`?

Reply:
> In general, I think there are duplicate code between this new method and the `runTx` method.
> 
> Is there room to refactor out common components? Or is the goal here to keep the `runTx` unmodified to keep the rebasing easier in the future? Or something else?

The goal was to keep the rebasing simple and also to cut the function down to what is needed for dYdX v4. 

> How do we plan to keep this method in sync w future updates to `runTx`?

Our e2e tests that execute through the ABCI++ flow should cover most of our needs but an audit of the differences in `runTx` versions during upgrades would help until we have more e2e tests.

Reply:
Makes sense. 

> audit of the differences in `runTx`

Do you think it's worth listing a checklist in [our readme?](https://github.com/dydxprotocol/cosmos-sdk#updating-cosmossdk-to-new-versions)

We also prob want to add an item to audit our `app/ante` impl, which we have been tracking as one offs using tickets like [this](https://linear.app/dydx/issue/CLOB-556), but I think it's better to create a checklist somewhere more visible